src/components/QuestionDisplay/QuestionDisplay.jsx

```diff
@@ -0,0 +1,94 @@
+import { useMemo, useState } from 'react';
+import {
+  Box,
+  Card,
+  CardContent,
+  LinearProgress,
+  Typography,
+} from '@mui/material';
+import AnswerButton from './AnswerButton';
+import useQuizContext from '../../hooks/useQuizContext';
+
+function QuestionDisplay() {
+  const { quizState, handleAnswerQuestion, handleCompleteQuiz } =
+    useQuizContext();
+  const [selectedAnswer, setSelectedAnswer] = useState(null);
+  const [showResult, setShowResult] = useState(false);
+
+  const currentQuestion = quizState.questions[quizState.currentQuestion];
+
+  const answers = [
+    currentQuestion.correct_answer,
+    ...currentQuestion.incorrect_answers,
+  ];
+
+  const shuffledAnswers = useMemo(() => {
+    setSelectedAnswer(null);
+    setShowResult(false);
+    return [...answers].sort(() => Math.random() - 0.5);
```

Comment:
This is difficult to read. Create additional variables to explain `() => Math.random() - 0.5`